src/api/roleApi.ts

```diff
@@ -71,18 +72,83 @@ export const assignUserRole = async (
     const data = {
       institution,
       username,
-      roles: [{ rolename, type: 'Role' }],
+      roles: [roleElement],
       type: 'User',
     };
+
     const response = await Axios.post(url, data, { headers, cancelToken });
     if (response.status === StatusCode.OK) {
       return response.data;
+    } else {
+      return { error: i18n.t('feedback:error.add_role') };
     }
   } catch (error) {
     if (error.response.status === StatusCode.CONFLICT) {
-      return { info: i18n.t('feedback:info.user_already_exists') };
+      // Update existing user instead of creating a new one
+      return await getUserAndAddRole(username, institution, roleElement, cancelToken);
     } else if (!Axios.isCancel(error)) {
       return { error: i18n.t('feedback:error.add_role') };
     }
   }
 };
```

Comment:
I ettertid ser jeg at dette sikkert kunne vært gjort mye enklere... hvis vi setter som krav at man må ha logget inn for å kunne bli satt til admin betyr det egentlig at vi bare trenger å forholde oss til PUT (aldri POST, siden backend automatisk vil opprette alle brukere som logger inn). Men dette er ting som fortsatt mangler på backend